lib/linux_admin/ssh.rb

```diff
@@ -0,0 +1,94 @@
+module LinuxAdmin
+  class SSH
+    attr_reader :ip
+    attr_reader :username
+    attr_reader :private_key
+    attr_reader :agent
+
+    def initialize(ip, username, private_key)
```

Comment:
The private key is only needed when using ssh agent. This is one of the reasons why I think we should be passing in the socket here like @bdunne suggested here https://github.com/ManageIQ/linux_admin/pull/160#issuecomment-232712525 from a previously initialized agent which the caller has set up explicitly.

Also it would also be good to have an optional password argument here in case the user does not have the auth keys set up.